packages/core-data/src/resolvers.js

```diff
@@ -525,3 +525,23 @@ export const getBlockPatternCategories =
 		} );
 		dispatch( { type: 'RECEIVE_BLOCK_PATTERN_CATEGORIES', categories } );
 	};
+
+export const getNavigationFallback =
+	() =>
+	async ( { dispatch } ) => {
+		const fallback = await apiFetch( {
+			path: '/wp-block-editor/v1/navigation-fallbacks?_embed',
+		} );
+
+		const record = fallback?._embedded?.self;
+
+		dispatch.receiveNavigationFallback( fallback );
```

Comment:
Should we pass the whole object here or just the id?

Reply:
In other words, I think we should hide the response's complexity from the consumer, the consumer only cares about the id or the object (which can be retrieved using getEntityRecord)

Reply:
I've gone back and forth over this one. Currently the block just requires the `id` and then independently fetches the object via a separate `getEntityRecord` call. This is because in situations where we are _not_ fetching a fallback the same `getEntityRecord` call simply fetches the Navigation referenced by the block's `ref` attribute. I wanted to avoid additional complexity in the `edit` code to handle setting the internal block state for the full Navigation post.

What I'm hearing you asking is should we just return the full object from this resolver (thus hiding the complexities of how that object is retrieved) and have different selectors for getting the `id` and the full Post. Did I understand that correctly?

Reply:
Not really, I'm asking whether the `useEffect` should be called again (setting the ref) if the "fallback id" changes or not.

Reply:
Trying to think of this as a user, it would be pretty disconcerting if the Navigation updated underneath you automatically. The goal is to have _some_ menu be available. There's an outside chance that some more "suitable" fallback gets created whilst you are editing the existing fallback and that process shouldn't cause you to have "magic" happen in the editor whilst you are working.

So the `useEffect` should _not_ be called again if the ID changes, meaning (I think) that we should "export" the function only from the `useSelect`.

Reply:
😄 Isn't that what I'm doing here?

```
// fallback is an ID
dispatch.receiveNavigationFallback( fallback );
```

Then separately we push the object entity into the existing reducer:

```
const record = fallback?._embedded?.self;
dispatch.receiveNavigationFallback( fallback );
```

This is why once the fallback endpoint is called we get the full post entity in state and when we (in other code) call `getEntityRecord('postType`, `wp_navigation`, %%FALLBACKID%%)` we don't see a network request.

It's been a long day so forgive me if I'm getting crossed wires.

Reply:
Yeah, it's a naming thing but also changing what the reducer stores. Something like

```
// fallback is an ID
dispatch.receiveNavigationFallbackId( fallback.id /* or fallback?._embedded?.id */ ) ;
```

and in the component you just do `select(coreStore).getNavigationFallbackId()` and you get an ID directly. 

It's the receiveEntityRecord that allows the `getEntityRecord` to work not the `receiveNavigationFallbackId` instead of a fallback object

Reply:
Currently calling `getNavigationFallback()` _will_ already get you an ID. The reducer stores an ID.

The reducer does _not_ store the full object. That is offloading to the existing entity reducer.

I think naming is really confusing things here. Let me do some renaming now.

Reply:
<img width="787" alt="Screenshot 2023-04-26 at 11 45 32 AM" src="https://user-images.githubusercontent.com/272444/234552372-bc807f11-eec9-4b2a-8e9d-74319f3f41b7.png">

Here's a screen of the call to the `getNavigationFallbackId` selector. As you can see it's not returning the "id", it's returning the whole response. So basically both the reducer and selector are returning the raw response from the API. I think they should only store and return the id.

Reply:
I literally just saw this in the code and then came back to this comment but you'd already responded 🤦 You are absolutely correct. Let me amend this.

Reply:
Ok that's all updated towards storing the ID only.